ze/tracer_ze_helpers.include.c

```diff
@@ -868,24 +901,93 @@ void readEnergy(uint32_t deviceIdx, uint32_t domainIdx, uint64_t *ts_us, uint64_
   *energy_uj = 0;
   zes_power_energy_counter_t energyCounter;
   if (zesPowerGetEnergyCounter(_sampling_hPowers[deviceIdx][domainIdx], &energyCounter) == ZE_RESULT_SUCCESS) {
-    // printf("Device %d - Power Domain %d: Total energy consumption: %lu Joules\n", deviceIdx, domainIdx, energyCounter.energy);
     *ts_us = energyCounter.timestamp;
     *energy_uj = energyCounter.energy;
   }
 }
 
+void readComputeE(uint32_t deviceIdx, computeEngineData *computeData ){
+  ze_result_t result;
+  for (uint32_t i = 0; i < _sampling_subDeviceCount; i++) {
+    computeData[i].computeActive = 0;
+    computeData[i].timestamp = 0;
+  }
+  for (uint32_t j = 0; j < _sampling_engineCounts[deviceIdx]; ++j) {
+    zes_engine_properties_t engineProp = {};
```

Comment:
```suggestion
    zes_engine_properties_t engineProp = {0};
    engineProp.stype = ZES_STRUCTURE_TYPE_ENGINE_PROPERTIES;
```
Per the documentation structures should be correctly typed before being used.